jerry-core/ecma/base/ecma-globals.h

```diff
@@ -690,15 +690,31 @@ typedef enum
 } ecma_string_container_t;
 
 /**
+ * ECMA string related constants
+ */
+typedef enum
+{
+  ECMA_STRING_CONTAINER_MASK = 0x7u, /**< container mask */
+  ECMA_STRING_REF_ONE = 1u << 3, /**< value for increasing or decreasing the reference counter */
+  ECMA_STRING_MAX_REF = 0x1fffu << 3, /**< maximum value of the reference counter (8191) */
+} ecma_string_constants_t;
```

Comment:
These are hard-wired constants here. Shouldn't we have some assert to ensure that the highest value in `ecma_string_container_t` fits in `ECMA_STRING_CONTAINER_MASK`? The use of `<< 3` is also related.

Moreover, IMHO this is an abuse of enums. These are just constants and the type `ecma_string_constants_t` is not used anywhere in the code. We do use enums with non-sequential values in the code already (i.e., for bit-fields), but always when the type itself is really used somewhere. I haven't seen enums used just for grouping values, as proposed here. (And I've vote against it.)